### ISSUE TITLE
Single-category swarmplot is broken with matplotlib 3.2

### ISSUE THREAD

**Issue by WMiller256** [mwaskom/seaborn#2013]
Running the first example from the [swarmplot documentation](https://seaborn.pydata.org/generated/seaborn.swarmplot.html), 

```
import seaborn as sns
sns.set(style="whitegrid")
tips = sns.load_dataset("tips")
ax = sns.swarmplot(x=tips["total_bill"])
```

With Python 3.x (tested 3.6.9 and 3.8.0) produces this
![SNSswarmplot-3 x](https://user-images.githubusercontent.com/33332811/78138248-7bc8a300-73db-11ea-96fd-6bfb689696f2.png)

but running it with Python 2.7 produces this
![SNSswarmplot-2 7](https://user-images.githubusercontent.com/33332811/78138249-7c613980-73db-11ea-9494-4ab8fa748612.png)

However, when specifying multiple categories, i.e.

```
import seaborn as sns
sns.set(style="whitegrid")
tips = sns.load_dataset("tips")
ax = sns.swarmplot(x=tips["total_bill"], y=tips["day"])
```

Python 3.x and Python 2.7 show the same plot:

![SNSswarmplot-4cat](https://user-images.githubusercontent.com/33332811/78138533-e7ab0b80-73db-11ea-8fe7-26aafeca0962.png)

so the inconsistency seems to be restricted to the case with a single category.

**Comment by mwaskom** [comment]
It’s unlikely that the python version matters. Are you using the same version of matplotlib and seaborn in each environment?

**Comment by WMiller256** [comment]
Should have included Matplotlib and Seaborn versions in the original post. With Python 3,x I'm using Matplotlib version 3.2.1 and Seaborn 0.10.0 and with Python 2.7 I'm using Matplotlib 2.2.5 and Seaborn 0.9.1.

**Comment by mwaskom** [comment]
I can replicate on matplotlib 3.2.1 but not 3.1.3 The 3.2.0 release notes say

> improvements to auto-scaling and auto-limits

So I imagine it has to do with those changes.

**Comment by mwaskom** [comment]
Changed the issue title because the Python version is a red herring.

**Comment by WMiller256** [comment]
That seems appropriate, I should have realized it's an issue with the Matplotlib version not with the Python version. Consequences of submitting an issue in the middle of the night I suppose. 

As far as the "improvements to auto-scaling and auto-limits" go, would explicitly setting the limits remedy the issue if that were the cause?

**Comment by mwaskom** [comment]
> As far as the "improvements to auto-scaling and auto-limits" go, would explicitly setting the limits remedy the issue if that were the cause?

No, `swarmplot` already [explicitly sets the limits before plotting](https://github.com/mwaskom/seaborn/blob/master/seaborn/categorical.py#L1361).

**Comment by mwaskom** [comment]
I think this is relevant: https://github.com/matplotlib/matplotlib/issues/17004#issuecomment-607961622

**Comment by ImportanceOfBeingErnest** [comment]
This one bisects to https://github.com/matplotlib/matplotlib/pull/13593. The solution would be to call `ax.autoscale_view()` in Line 1426 https://github.com/mwaskom/seaborn/blob/b47d84383b4c42e59eea7a36bd7672dd5ac79506/seaborn/categorical.py#L1426
since you only set the xlimits explicitely and the ylimits would need to autoscale before using a frozen transform (which any `.inverted()` transform is).

**Comment by mwaskom** [comment]
Fixed in #2017 thanks @ImportanceOfBeingErnest
